extension/observer/kafkatopicsobserver/extension.go

```diff
@@ -24,35 +26,116 @@ var (
 
 type kafkaTopicsObserver struct {
 	*observer.EndpointsWatcher
-	logger *zap.Logger
-	config *Config
-	cancel func()
-	once   *sync.Once
-	_      context.Context
+	logger     *zap.Logger
+	config     *Config
+	cancel     func()
+	once       *sync.Once
+	ctx        context.Context
```

Comment:
Can I suggest instead of storing the context and the once operation that you start up a background routine using https://pkg.go.dev/golang.org/x/sync/errgroup#Group ?

The reason being is that this can manage the routine for you and similar to a threadpool in other languages. 

You can store the cancel in the type so you can shutdown the group at any point.

Reply:
Thanks for noticing that. Btw, I've seen that context is stored in `dockerObserver` type, so maybe it would be good to change it in that observer too. 

For this kafka topics observer instead of errorgroup I adopted similar approach as in `cfGardenObserver`, which is storing `doneChan` inside the observer struct and closing it in the `Shutdown` method. What do you think about it?